Source/hydro/fourth_order.F90

```diff
@@ -93,6 +93,28 @@ subroutine states(idir, &
                       a_int(i,j,k) = (1.0_rt/12.0_rt)*(25.0_rt*a(i-1,j,k,n) - 23.0_rt*a(i-2,j,k,n) + &
                                                        13.0_rt*a(i-3,j,k,n) - 3.0_rt*a(i-4,j,k,n))
 
+                   else if (i == domlo(1)-1 .and. physbc_lo(1) == Outflow) then
+                      ! extrapolate to the domlo(1)-1 cell using a
+                      ! conservative cubic polynomial averaged over
+                      ! the cell
+                      ac = 4.0_rt*a(domlo(1),j,k,n) - 6.0_rt*a(domlo(1)+1,j,k,n) + 4.0_rt*a(domlo(1)+2,j,k,n) - a(domlo(1)+3,j,k,n)
+
+                      ! now use the 1-sided stencil from above with
+                      ! this extrapolated value
+                      a_int(i,j,k) = (1.0_rt/12.0_rt)*(25.0_rt*ac - 23.0_rt*a(i+1,j,k,n) + &
+                                                       13.0_rt*a(i+2,j,k,n) - 3.0_rt*a(i+3,j,k,n))
+
+                   else if (i == domhi(1)+2 .and. physbc_hi(1) == Outflow) then
+                      ! extrapolate to the domhi(1)+1 cell using a
+                      ! conservative cubic polynomial averaged over
+                      ! the cell
+                      ac = 4.0_rt*a(domhi(1),j,k,n) - 6.0_rt*a(domhi(1)-1,j,k,n) + 4.0_rt*a(domhi(1)-2,j,k,n) - a(domhi(1)-3,j,k,n)
```

Comment:
The check here is on `i == domhi(1)+2` but we're using data starting at `domhi(1)`, not `domhi(1)+1` -- is this right?

Reply:
Yes, but this is a little tricky.  When we limit, then to get the right state for interface domhi+1, we need to have both of the interface states that cell domhi+1 sees, so this is domhi+1/2.R and domhi+3/2,L, which would have indices in the code domhi+1 and domhi+2 respectively.  So for outflow, I decided to extrapolate using a 4th order conservative interpolant to get that outer interface.